### ISSUE TITLE
storybook: `@default` is ignored for story doc table rendering since 6.3

### ISSUE THREAD

**Issue by Hotell** [microsoft/fluentui#19292]
We have following story definition (that includes `@default` JSDoc pragma):

```tsx
// @filename Default.stories.tsx

interface Props {
  /**
   *
   * @default small
   */
  size?: 'small' | 'medium' | 'large';
}
export const Default = (args: Props) => {
   
  return <div>dummy {args.size}</div>
}

export default {
 title: 'Example/Default',
 component: Default
}
```

**Current Behaviour**

- when using `Controls` to autogenerate storybook controls from component interfaces, `@default` is not showing/stopped working

![2021-08-06 at 13 01](https://user-images.githubusercontent.com/1223799/128500982-213bf0ea-9b6e-41a9-8eb4-5bd35a33c27f.png)

**Expected behaviour**

- `@default` works as expected/before and properly renders/assigns value to storybook component/ storybook table
![2021-08-06 at 13 03](https://user-images.githubusercontent.com/1223799/128501253-0f6b5022-792b-4262-a0ab-3b343032d515.png)

### Additional troubleshooting

- spawned CRA with storybook  (6.3.) -> ✅ works as expected
  - `npx create-react-app storybook-docgen-test --template typescript`
  - `cd  storybook-docgen-test && npx sb init`
  - modify `Button.tsx` to use `@default` pragma instead of default args pattern
- our repo:
  - doesn't work
  - works only with manual `argTypes`
  
```tsx
Default.argTypes = {
  size: {
    name: 'size',
    type: { name: 'string', required: false },

    table: {
      defaultValue: { summary: 'small' },
    },
  },
};
```
- havent found any open tickets on storybook repo

**Comment by TristanWatanabe** [comment]
@Hotell This issue started occurring when storybook upgraded their `react-docgen-typescript` dependency to `^2.0.0` ([PR here](https://github.com/storybookjs/storybook/pull/15247)) and released in version `6.3.0-rc.7`. There might be some underlying conflict with `react-docgen-typescript` that's causing the defaultValue to be null with our repo (see screenshots below). Others have also come across this very same regression (see this [comment](https://github.com/storybookjs/storybook/issues/15378#issuecomment-879027816)). However, the only usage we have of `react-docgen-typescript` in our repo is in conformance tests so I'm not sure this is it unless I'm missing something. I considered that using `storybook/builder-webpack5` might be an issue but converged packages are still running on `storybook/builder-webpack4` since they have not been opted in to use `storybook/builder-webpack5` yet so that's probably not it. Just wanted to get your thoughts on this.

**Our Repo** (defaultValue is null):
![image](https://user-images.githubusercontent.com/8649804/128583545-8463d7e4-18a6-4e87-81b0-1595d04877f4.png)


**Create React App** (defaultValue is properly applied):
![image](https://user-images.githubusercontent.com/8649804/128583534-5bc4c422-7521-475a-b85f-25b4a1596760.png)

**Comment by Hotell** [comment]
fantastic investigation @TristanWatanabe  🙌

**I'd suggest following action:**
- lets use that temporary yarn resolutions field as a quick fix
- lets keep this issue open to track when it will be resolved in both react-docgen and storybook (then we can update versions - remove resolutions and close this)
- create issue to migrate conformance package to latest react-docgen

**Comment by ecraig12345** [comment]
Sort of copying this comment https://github.com/microsoft/fluentui/issues/19329#issuecomment-911832891

`react-docgen-typescript` 2 depends on `typescript` 4.3, so until we move to TS 4.3 or there's a fix restoring compat in `react-docgen-typescript` (which would actually be quite easy, see https://github.com/styleguidist/react-docgen-typescript/issues/366#issuecomment-906631869) we have to stick with v1.

**Comment by ecraig12345** [comment]
This should be fixed by the upgrade to typescript 4.3 / react-docgen-typescript 2 I think. #20909
